oauth2/token_hook.go

```diff
@@ -166,12 +196,28 @@ func TokenHook(reg interface {
 			return nil
 		}
 
+		var clientMeta json.RawMessage
+		if hydraClient, ok := requester.GetClient().(*client.Client); ok {
+			clientMeta = json.RawMessage(hydraClient.Metadata)
+		} else {
+			clientMeta = []byte("null")
+		}
+
+		// Always include client authentication parameters, and any sensitive parameters for the requested grant types.
+		sensitiveParameters := append([]string{"client_secret", "client_assertion"}, sensitiveAccessRequestParametersForGrantTypes(requester.GetGrantTypes()...)...)
+
+		payload := maps.Clone(requester.GetRequestForm())
+		// Delete all sensitive parameters from the access request form before sending it to the hook.
+		maps.DeleteFunc(payload, func(s string, _ []string) bool {
+			return slices.Contains(sensitiveParameters, s)
+		})
 		request := Request{
 			ClientID:        requester.GetClient().GetID(),
+			ClientMetadata:  clientMeta,
```

Comment:
I mean if we have the client it probably makes sense to just send the client instead.